sdk/src/logs/simple_log_processor.cc

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/simple_log_processor.h"
+
+#include <chrono>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * Initialize a simple log processor.
+ * @param exporter the configured exporter where log records are sent
+ */
+SimpleLogProcessor::SimpleLogProcessor(std::unique_ptr<LogExporter> &&exporter)
+    : exporter_(std::move(exporter))
+{}
+
+/**
+ * Batches the log record it receives in a batch of 1 and immediately sends it
+ * to the configured exporter
+ */
+void SimpleLogProcessor::OnReceive(
+    std::unique_ptr<opentelemetry::logs::LogRecord> &&record) noexcept
+{
+  std::vector<std::unique_ptr<opentelemetry::logs::LogRecord>> batch;
+  batch.emplace_back(std::move(record));
+
+  // Get lock to ensure Export() is never called concurrently
+  std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+
+  if (exporter_->Export(batch) == ExportResult::kFailure)
+  {
+    /* TODO: alert user of the failed or timedout export result */
```

Comment:
Should failure of the exporter be reported *by the exporter* or by the log processor?

I think the log processor should report its own failure and let the Exporter report its own failure as well.

Reply:
I'm not sure about this, but these language repos seem to report the exporter's shutdown as part of the processor's shutdown result as well: 
* Dotnet: https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry/BatchExportProcessor.cs#L184
* Java: https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java#L123

Reply:
Gotcha.  The real issue here is that *theoretically* you are the logging framework, so how do you alert the user of failure if you're already failing to send out logs.

Note:  My meta-comment is I'm trying to understand if your TODO has a solution or not.  I think maybe it'd be useful to open a bug/ticket discussing the need to report Failures in the SDK in a consistent way that can work around issues in  Exporting Telemetry.   E.g. we don't want to *log* a failure if we are the logging framework and that failure would disappear because we are the ones failing.

I don't remember the specification having any specifics on how to handle failures of this type in a way that users can find + diagnose the problem.   I'm just trying to figure out how big this TODO is, and I think I'd love to have a bug open about it so we can start a (broader) discussion.

Reply:
FYI - self diagnostics is briefly covered by the spec https://github.com/open-telemetry/opentelemetry-specification/blob/609a63784b2da7d59efe2a39822a23b09ae5b637/specification/error-handling.md#self-diagnostics.

Reply:
In OpenTelemetry .NET, folks are using a simple custom logging API (which is not part of ILogger) and route data to a memory mapped file https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry/Internal/SelfDiagnostics.cs.

Another way of doing this is via ZPages, which is an experimental project though.

Reply:
Good point @jsuereth . This is something we need to define for our project, and needs broader discussion - either using callback mechanism or providing a ErrorHandler which can be used throughout the project.  I will create a ticket for this.

Reply:
Created #408. Let's discuss there. Thought this would be out of scope for this PR.